harness/api_checkers.go

```diff
@@ -0,0 +1,437 @@
+	span.Finish()
+}
+
+// TestStartSpanWithParent checks if a Tracer can start a span with a specified parent.
+func (s *APICheckSuite) TestStartSpanWithParent() {
+	parentSpan := s.tracer.StartSpan("parent")
+	assert.NotNil(s.T(), parentSpan)
+
+	span := s.tracer.StartSpan(
+		"Leela",
+		opentracing.ChildOf(parentSpan.Context()))
+	span.Finish()
+
+	span = s.tracer.StartSpan(
+		"Leela",
+		opentracing.FollowsFrom(parentSpan.Context()),
+		opentracing.Tag{Key: "birthplace", Value: "sewers"})
+	if s.opts.Probe != nil {
+		assert.True(s.T(), s.opts.Probe.SameTrace(parentSpan, span))
+	}
+	span.Finish()
+
+	parentSpan.Finish()
+}
+
+// TestSetOperationName attempts to set the operation name on a span after it has been created.
+func (s *APICheckSuite) TestSetOperationName() {
+	span := s.tracer.StartSpan("").SetOperationName("Farnsworth")
+	span.Finish()
+}
+
+// TestSpanTagValueTypes sets tags using values of different types.
+func (s *APICheckSuite) TestSpanTagValueTypes() {
+	span := s.tracer.StartSpan("ManyTypes")
+	span.
+		SetTag("an_int", 9).
+		SetTag("a_bool", true).
+		SetTag("a_string", "aoeuidhtns")
+}
+
+// TestSpanTagsWithChaining tests chaining of calls to SetTag.
+func (s *APICheckSuite) TestSpanTagsWithChaining() {
+	span := s.tracer.StartSpan("Farnsworth")
+	span.
+		SetTag("birthday", "9 April, 2841").
+		SetTag("loves", "different lengths of wires")
+	span.
+		SetTag("unicode_val", "non-ascii: \u200b").
+		SetTag("unicode_key_\u200b", "ascii val")
+	span.Finish()
+}
+
+// TestSpanLogs tests calls to log keys and values with spans.
+func (s *APICheckSuite) TestSpanLogs() {
+	span := s.tracer.StartSpan("Fry")
+	span.LogKV(
+		"event", "frozen",
+		"year", 1999,
+		"place", "Cryogenics Labs")
+	span.LogKV(
+		"event", "defrosted",
+		"year", 2999,
+		"place", "Cryogenics Labs")
+
+	ts := time.Now()
+	span.FinishWithOptions(opentracing.FinishOptions{
+		LogRecords: []opentracing.LogRecord{
+			{
+				Timestamp: ts,
+				Fields: []log.Field{
+					log.String("event", "job-assignment"),
+					log.String("type", "delivery boy"),
+				},
+			},
+		}})
+
+	// Test deprecated log methods
+	span.LogEvent("an arbitrary event")
+	span.LogEventWithPayload("y", "z")
+	span.Log(opentracing.LogData{Event: "y", Payload: "z"})
+}
+
+func assertEmptyBaggage(t *testing.T, spanContext opentracing.SpanContext) {
+	if !assert.NotNil(t, spanContext, "assertEmptyBaggage got empty context") {
+		return
+	}
+	spanContext.ForeachBaggageItem(func(k, v string) bool {
+		assert.Fail(t, "new span shouldn't have baggage")
+		return false
+	})
+}
+
+// TestSpanBaggage tests calls to set and get span baggage, and if the CheckBaggageValues option
+// is set, asserts that baggage values were successfully retrieved.
+func (s *APICheckSuite) TestSpanBaggage() {
+	span := s.tracer.StartSpan("Fry")
+	assertEmptyBaggage(s.T(), span.Context())
+
+	spanRef := span.SetBaggageItem("Kiff-loves", "Amy")
+	assert.Exactly(s.T(), spanRef, span)
```

Comment:
I thought Suite had all those assert methods on itself, i.e. instead of passing `s.T()` you can just do `s.Equals()`